onetl/db/db_reader/db_reader.py

```diff
@@ -501,6 +502,47 @@ def validate_options(cls, options, values):
 
         return None
 
+    @slot
+    def has_data(self) -> bool:
+        """Returns ``True`` if there is some data in the source, ``False`` otherwise.
+
+        .. warning::
+            This method should be used within the context of :ref:`strategy` if you use :etl-entities:`hwm <hwm/index.html>`. It may not work as expected when used outside of a strategy context. This does not apply to :ref:`snapshot-strategy`, where the method functions normally.
+        """
```

Comment:
Что-то тут как-то сложно написано. Оно не просто `may not work as expected`, тут исключение вылетает.

Предлагаю так:
```suggestion
        """Returns ``True`` if there is some data in the source, ``False`` otherwise.

        .. warning::

            If :etl-entities:`hwm <hwm/index.html>` is used, then methods should be called inside :ref:`strategy` context. And vise-versa, if HWM is not used, this method should not be called within strategy.

        Raises
        ------
        RuntimeError
            Current strategy is not compatible with HWM parameter.

        Example
        -------
        
        .. code:: python

            ...
        """
```

У `.run()`, `.has_data()` и `.raise_if_no_data()` должны быть одинаковые примечания насчет использования с HWM.

Reply:
И добавь небольшой пример использования каждому методу. Создание объекта DBReader в примере можно опустить